cf_xarray/accessor.py

```diff
@@ -2335,6 +2363,82 @@ def bounds_to_vertices(
                 )
         return obj
 
+    @property
+    def grid_mappings(self) -> dict[str, list[str]]:
+        """
+        Property that returns a dictionary mapping keys
+        to the variable names of their grid_mapping.
+
+        Returns
+        -------
+        dict
+            Dictionary mapping keys to the variable names of their grid_mapping.
+
+        See Also
+        --------
+        Dataset.cf.get_grid_mapping_name
+
+        Examples
+        --------
+        >>> from cf_xarray.datasets import mollwds
+        >>> mollwds.cf.bounds
+        {'lat': ['lat_bounds'], 'latitude': ['lat_bounds'], 'lon': ['lon_bounds'], 'longitude': ['lon_bounds']}
+        """
+
+        obj = self._obj
+        keys = self.keys() | set(obj.variables)
+
+        vardict = {
+            key: self._drop_missing_variables(
+                apply_mapper(_get_grid_mapping, obj, key, error=False)
+            )
+            for key in keys
+        }
+
+        return {k: sorted(v) for k, v in vardict.items() if v}
+
+    def get_grid_mapping(self, key: str) -> DataArray | Dataset:
+        """
+        Get grid_mapping variable corresponding to key.
+
+        Parameters
+        ----------
+        key : str
+            Name of variable whose grid_mapping is desired
+
+        Returns
+        -------
+        DataArray
+        """
+
+        results = self.grid_mappings.get(key, [])
+        if not results:
+            raise KeyError(f"No results found for {key!r}.")
+
+        return self._obj[results[0] if len(results) == 1 else results]
+
+    def get_grid_mapping_name(self, key: str) -> str:
+        """
+        Get bounds dim name for variable corresponding to key.
```

Comment:
Let's update this docstring.


In general, this one seems redundant (`ds.cf.get_grid_mapping('temp").attrs["grid_mapping_name"]` seems not too bad) but happy to defer to an actual user :)

Reply:
That's true! I just thought, having this function explictily in the API would raise some awareness for the difference between `grid_mapping` and `grid_mapping_name` attributes (especially if they are different)!